File: 2111-minimum-operations-to-make-the-array-k-increasing/2111-minimum-operations-to-make-the-array-k-increasing.py
class Solution:
    def find_ans(self , a) : 
        lis = []
        for i in a :
            idx = bisect_right(lis,i ); 
            if idx < len(lis):
                lis[idx] =i 
            else:
                lis.append(i);
        return len(lis) 
            
    def kIncreasing(self, a: List[int], k: int) -> int:
        n = len(a); 
        ans =0 ; 
        for i in range(k) :
            b = [a[j] for j in range(i , n , k )]
            ans += len(b) - self.find_ans(b)
        return ans

File: 2111-minimum-operations-to-make-the-array-k-increasing/2111-minimum-operations-to-make-the-array-k-increasing.cpp
class Solution {
public:
    int find_ans(vector<int>&a){
        vector<int>lis; 
        for(auto i: a){
            auto idx = upper_bound(lis.begin() , lis.end() , i)-lis.begin(); 
            if(idx < lis.size()){
                lis[idx] = i ; 
            }else{
                lis.push_back(i); 
            }
        }
        return lis.size(); 
    }
    int kIncreasing(vector<int>& a, int k) {
        int n = a.size(); 
        int ans = 0  ;
        for( int i =0 ;i < k ;i++){
            vector<int>b; 
            for(int j = i ; j < n ; j+=k ){
                b.push_back(a[j]); 
            }
            ans+= b.size() - find_ans(b);
        }
        return ans; 
    }
};